### PR TITLE
Add new plugin : Genkit HNSW

### PR DESCRIPTION
**Pull request by retzd-tech**
_Before you submit a pull request, please make sure you have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md)._

**This pull request is related to:**

- [ ] A bug
- [x] A new feature
- [x] Documentation
- [ ] Other (please specify)

**I have checked the following:**

- [x] I have read and understood the [contribution guidelines](https://github.com/TheFireCo/genkit-plugins/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/TheFireCo/genkit-plugins/blob/main/CODE_OF_CONDUCT.md);
- [x] I have added new tests (for bug fixes/features);
- [x] I have added/updated the documentation (for bug fixes / features).

**Description:**
Add a new plugin Genkit HNSW

### CODE REVIEW

**Comment by EPMatt** [comment]
@retzd-tech thanks for the contribution! 🎉  Will review and check what's missing at some point today

File: plugins/hnsw/LICENSE
Comment (EPMatt) [inline plugins/hnsw/LICENSE:1]: @davidoort needs to be updated to The Fire Company license before publishing

File: plugins/hnsw/README.md
Comment (EPMatt) [inline plugins/hnsw/README.md:1]: @davidoort we should align this to our README template

File: plugins/hnsw/coverage/clover.xml
Comment (EPMatt) [inline plugins/hnsw/coverage/clover.xml:1]: I believe these coverage files do not need to be committed at all.

File: plugins/hnsw/CONTRIBUTING.md
Comment (EPMatt) [inline plugins/hnsw/CONTRIBUTING.md:1]: @davidoort the content of this file should be translated to GitHub issues so people can easily spot the missing contributions and help with that. 👍🏻

File: plugins/hnsw/package.json
Comment (EPMatt) [inline plugins/hnsw/package.json:39]: @davidoort I moved `@genkit-ai/ai` and `@genkit-ai/core` to peer dependencies, not sure if some of these need to be moved there as well. Maybe worth checking.

File: plugins/hnsw/tsconfig.json
Comment (EPMatt) [inline plugins/hnsw/tsconfig.json:1]: @davidoort let's also align this to the ones we are using on other plugins

**Review by EPMatt — CHANGES_REQUESTED** [review]
@retzd-tech thank you so much for your contribution! Fixed some dependencies issues here and there, now the build process completes successfully.

@davidoort I suggested some changes, tagging you in the comments. I'd suggest to take those steps before merging. Let me know if you need help.

Given the size of the PR, I advise to first merge this and publish the plugin, then work  on the other needed changes (Code Owners, docs, etc.) on a different PR.

File: plugins/hnsw/LICENSE
Comment (davidoort) [reply plugins/hnsw/LICENSE:1]: The other plugins don't have a license file so I'm removing it from this one as well

File: plugins/hnsw/CONTRIBUTING.md
Comment (davidoort) [inline plugins/hnsw/CONTRIBUTING.md:1]: This is a good idea for some of the other plugins!

Comment (davidoort) [reply plugins/hnsw/CONTRIBUTING.md:1]: this is a good idea, but not required for merging imo

**Review by davidoort — APPROVED** [review]
Thank you so so much for the contribution and addressing all comments @retzd-tech ! I'm so excited to get this released! 🎉🔥

**Pull request merged**
